packages/xod-fs/src/load.js

```diff
@@ -122,35 +122,48 @@ export const loadProjectWithoutLibs = projectPath => R.composeP(
   readDir
 )(projectPath);
 
-// :: Path -> Path -> Path -> Promise [File] Error
-export const loadProjectWithLibs = (projectPath, workspace, libWorkspace = workspace) => {
-  const fullProjectPath = resolvePath(path.resolve(workspace, projectPath));
-  const libPath = resolvePath(libWorkspace);
+// :: Path -> Path -> Path -> Path -> Promise [File] Error
+export const loadProjectWithLibs = R.curry(
+  (addWorkspaceList, projectPath, workspace) => {
+    const fullProjectPath = resolvePath(path.resolve(workspace, projectPath));
+    const libPath = resolvePath(workspace);
 
-  return Promise.all([
-    loadProjectWithoutLibs(fullProjectPath),
-    loadAllLibs(libPath),
-  ]).then(([projectFiles, libs]) => ({ project: projectFiles, libs }))
-    .catch(err => Promise.reject(
-      Object.assign(err, {
-        libPath,
-        fullProjectPath,
-        workspace,
-      })
-    ));
-};
+    const workspacePaths = R.compose(
+      R.concat([libPath]),
+      R.map(resolvePath)
+    )(addWorkspaceList);
+
+    return Promise.all([
+      loadProjectWithoutLibs(fullProjectPath),
+      loadLibsFromWorkspaceList(workspacePaths),
+    ]).then(([projectFiles, libs]) => ({ project: projectFiles, libs }))
+      .catch(err => Promise.reject(
+        Object.assign(err, {
+          libPath,
+          fullProjectPath,
+          workspace,
+        })
+      ));
+  }
+);
 
 // :: Path -> Promise Project Error
-//
-// Loads a regular XOD project placed in a workspace. The workspace and project
-// name are determined by path provided. It is expected to be a path to the
-// project directory, e.g. `/path/to/workspace/my-proj`.
-//
-// Returns a Promise of complete `Project` (see `xod-project`).
-export const loadProject = projectPath =>
+/**
+ * Loads a regular XOD project placed in a workspace. The workspace and project
+ * name are determined by path provided. It is expected to be a path to the
+ * project directory, e.g. `/path/to/workspace/my-proj`.
+ *
+ * Also it loads a libs from additional workspace, that contains specified project.
+ * If `addWorkspaceList` is not nil and empty, it will load libs from this list of workspaces too.
+ * If all workspaces have libs with same names, will be loaded only one —
+ * from local workspace, that contains project (it has highest priority).
+ *
+ * Returns a Promise of complete `Project` (see `xod-project`).
+ */
+export const loadProject = (projectPath, addWorkspaceList = []) =>
```

Comment:
Ough. The comment should be grammar-checked, too many mistakes.

`add` is a verb, use `additional` or better `extra`… `extraLibDirs`

I see no point in the `workspace` medium. Could it operate on library directories directly? It would make things easier to understand and maintain.